### DISCUSSION TITLE
Can't import a module using 'mip', (VSCODE/Pico W/micropython), but works with pip and VSCODE Python

### DISCUSSION THREAD

**Discussion by breezin320** [micropython/micropython#12064] [Libraries & Drivers]
I am trying to import 'untangle' to use on a Pico W; am using VSCODE, Pico-W-Go.  I can import 'untangle' using pip, and the module is found when running Python in VSCODE and import.  However, when I use REPL, and 'mip' install 'untangle', it reports not found.  Here's the result:

>>> mip.install("untangle")
Installing untangle (latest) from https://micropython.org/pi/v2 to /lib
Package not found: https://micropython.org/pi/v2/package/6/untangle/latest.json
Package may be partially installed

I tried looking at the links directly, but they either don't exist, or are forbidden.  I'm pretty new at this, so appreciate the help and guidance!

best,
dave

**Comment by ricksorensen** [comment]
`mip` does not use the PyPi data base like `pip`.  Does this answer from developers help?
[https://github.com/orgs/micropython/discussions/9565](https://github.com/orgs/micropython/discussions/9565)

**Comment by bidrohini** [comment]


**Reply by mattytrentini** [reply]
@bidrohini it appears that you are regularly using ChatGPT - or some other LLM - to produce answers to questions. 😛  If you're going to insist on doing so then please review and edit the response for correctness and trim for brevity. So far, the suspicious responses have been a mix of incorrect, confusing and misleading...with an occasional useful nugget. If you're unsure whether the answer is correct then please refrain from posting.

**Comment by scruss** [comment]
by untangle, do you mean this: [stchris/untangle: Converts XML to Python objects](https://github.com/stchris/untangle)?

If so, this won't work with MicroPython: it relies on xml.sax, which isn't ported to MicroPython and would probably run out of memory on most ports

**Comment by breezin320** [comment]
Thanks, all, for your responses!  I think using an approach other than 'mip' and what's available in the micropython library might trip over a bunch of issues, such as missing dependencies, improper imports, etc.  I am such a beginner that these would take me down a rabbit hole that I'd never be able to crawl out of.....

The original issue was to find a way to parse an XML website page.   This point was made by SCRUSS, the xml libraries might be too huge to really work on a PICO.  Fortunately, we have some tools with regex (but not .findall! :-)), and the page is straightforward, that a site-specific parser could work...

thanks again to all for your help, advice and guidance....much appreciated!
